### ISSUE TITLE
Default Finalizer Identifier

### ISSUE THREAD

**Issue by NArnott** [buehler/dotnet-operator-sdk#339]
**Describe the bug**
The default Finalizer identifier potentially duplicates finalizer in the name. 

**To Reproduce**
Create a class called DemoFinalizer and implement IResourceFinalizer. By default, IResourceFinalizer.Identifier will produce the following identifier:
`some.group.com/demofinalizerfinalizer`

**Expected behavior**
I'd expect that if the class is appended with "Finalizer", which is a standard naming convention, the identifier would be `.../demofinalizer`. If the class doesn't end with "Finalizer", then it should append "Finalizer" to the end.

In short:
Class Name `DemoFinalizer` should produce `some.group.com/demofinalizer` 
Class Name `DemoOther` should produce `some.group.com/demootherfinalizer`

**Comment by buehler** [comment]
Hey @NArnott

Thanks for the report :-) This seems like a good thing to take care of. Thanks!
